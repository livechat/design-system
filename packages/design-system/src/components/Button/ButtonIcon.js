import * as React from 'react';
import * as PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import getMergedClassNames from '../../utils/getMergedClassNames';
import styles from './style.scss';

const cx = classNames.bind(styles);

export const ButtonIcon = ({ position, children }) => {
  const baseClass = 'btn__icon';
  const classes = getMergedClassNames(
    cx({
      [`${baseClass}`]: true,
      [`${baseClass}-left`]: position === 'left',
      [`${baseClass}-right`]: position === 'right'
    })
  );
  return <i className={classes}>{children}</i>;
};

ButtonIcon.propTypes = {
  position: PropTypes.oneOf(['left', 'right']).isRequired,
  children: PropTypes.node.isRequired
};
