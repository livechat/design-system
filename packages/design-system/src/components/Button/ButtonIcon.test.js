import * as React from 'react';
import { shallow } from 'enzyme';
import { ButtonIcon } from './ButtonIcon';

describe('ButtonIcon', () => {
  it('should render button with positioned icon', () => {
    const IconMock = () => <div />;
    const leftIconButton = shallow(
      <ButtonIcon position="left">
        <IconMock />
      </ButtonIcon>
    );
    const rightIconButton = shallow(
      <ButtonIcon position="right">
        <IconMock />
      </ButtonIcon>
    );

    expect(leftIconButton).toMatchSnapshot();
    expect(rightIconButton).toMatchSnapshot();
  });
});
