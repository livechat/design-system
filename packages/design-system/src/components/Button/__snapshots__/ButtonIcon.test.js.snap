// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonIcon should render button with positioned icon 1`] = `
<i
  className="btn__icon btn__icon-left"
>
  <IconMock />
</i>
`;

exports[`ButtonIcon should render button with positioned icon 2`] = `
<i
  className="btn__icon btn__icon-right"
>
  <IconMock />
</i>
`;
